src/components/buttons/HdLoaderButton.vue

```diff
@@ -1,54 +1,93 @@
-      </button>
+  <div
+    :class="loaderButtonClasses"
+    :style="{
+      height: `${buttonHeight}px`,
+    }"
+    class="loaderButton"
+    @keydown="setUsingMouse(false)"
+    @mousedown="setUsingMouse(true)"
+  >
+    <button
+      ref="button"
+      :style="buttonStyles"
+      class="btn btn--primary loaderButton__button"
+      @click="clicked"
+      @keyup.enter.space="clicked"
+      @transitionend="runTransitionQueue"
+      :disabled="disabled"
+    >
+      <span class="loaderButton__button__label">{{ label }}</span>
+    </button>
 
-      <!-- Loading circle -->
-      <svg class="loaderButton__svg loaderButton__svg--progressCircle" v-show="state == 'loading'" :width="`${buttonHeight}px`" :height="`${buttonHeight}px`" ref="circle">
-        <path
-          :d="circlePath"
-          :style="{strokeWidth: `${loadingCircleStrokeWidth}px`}"
-        />
-      </svg>
+    <!-- Loading circle -->
+    <svg
+      v-show="state == 'loading'"
+      ref="circle"
+      :width="`${buttonHeight}px`"
+      :height="`${buttonHeight}px`"
+      class="loaderButton__svg loaderButton__svg--progressCircle"
+    >
+      <path
+        :d="circlePath"
+        :style="{strokeWidth: `${loadingCircleStrokeWidth}px`}"
+      />
+    </svg>
 
-      <svg class="loaderButton__svg loaderButton__svg--checkmark" width="46px" height="46px" ref="check">
-        <!-- Checkmark -->
-        <g stroke="#1895FF" :style="scaledPath">
-          <path d="m11.066433,23.18223l8.364397,7.766946"/>
-          <path d="m34.933565,15.050822l-15.180095,15.533867"/>
-        </g>
-      </svg>
+    <svg
+      ref="check"
+      class="loaderButton__svg loaderButton__svg--checkmark"
+      width="46px"
+      height="46px"
+    >
+      <!-- Checkmark -->
+      <g
+        :style="scaledPath"
+        stroke="#2988FF"
+      >
+        <path d="m11.066433,23.18223l8.364397,7.766946" />
+        <path d="m34.933565,15.050822l-15.180095,15.533867" />
+      </g>
+    </svg>
 
-      <!-- Error cross -->
-      <svg class="loaderButton__svg loaderButton__svg--cross" width="46px" height="46px" ref="cross">
-        <g stroke="#1895FF" :style="scaledPath">
-          <path d="m23,23l-10.3,-10.3"/>
-          <path d="m23,23l10.3,10.3"/>
-          <path d="m23,23l-10.3,10.3"/>
-          <path d="m23,23l10.3,-10.3"/>
-        </g>
-      </svg>
+    <!-- Error cross -->
+    <svg
+      ref="cross"
+      class="loaderButton__svg loaderButton__svg--cross"
+      width="46px"
+      height="46px"
+    >
+      <g
+        :style="scaledPath"
+        stroke="#2988FF"
+      >
+        <path d="m23,23l-10.3,-10.3" />
+        <path d="m23,23l10.3,10.3" />
+        <path d="m23,23l-10.3,10.3" />
+        <path d="m23,23l10.3,-10.3" />
+      </g>
+    </svg>
   </div>
 </template>
 
 <script>
 import { getRandomInt, circleToPath } from 'homeday-blocks/src/services/utils';
 import debounce from 'lodash/debounce';
 
+const LOADING_STATE = Object.freeze({
```

Comment:
👍 

I would suggest you to export this constant so whenever someone imports it they can use the Enum as well.
Similar to what we've here:
- https://github.com/homeday-de/homeday-blocks/blob/develop/src/components/buttons/HdButton.vue (You can also `Object.freeze` this guy hehehe)
- https://github.com/homeday-de/homeday-blocks/blob/develop/main.js#L20

Reply:
Well, that's how I implement an Enum in JS because what you meant is just a constant 😂